### DISCUSSION TITLE
Getting the length of a horizontal bar after animation

### DISCUSSION THREAD

**Discussion by thoughtassassin** [chartjs/chartjs-plugin-datalabels#320] [Q&A] [answered]
I created a [Stack Overflow](https://stackoverflow.com/questions/72452650/how-do-you-get-the-width-of-a-datalabel-from-the-chartjs-plugin-after-the-chart) with this question. Here is the [codepen](https://codepen.io/thoughtassassin/pen/KKQZEXN?editors=1010) that I am working on. I want to get the width of the bar in the offset function after the chart animates. Is there a way to do that? Currently, with the animation on the width returns `NaN`.

**Comment by stockiNail** [comment, marked as answer]
@thoughtassassin I think you should use [`getProps`](https://www.chartjs.org/docs/latest/api/interfaces/BarElement.html#getprops) method of data element.

Here is the doc  from CHART.JS:

```javascript
  /**
   * Gets the current or final value of each prop. Can return extra properties (whole object).
   * @param {string[]} props - properties to get
   * @param {boolean} [final] - get the final value (animation target)
   * @return {object}
   */
  getProps(props, final) {
   ... 
  }
}
```
To get the value of width at the end of animation, have a look to following code, taken from you codepen:

```javascript
offset: function(context) {
  const chart = context.chart;
  const area = chart.chartArea;
  const meta = chart.getDatasetMeta(context.datasetIndex);
  const model = meta.data[context.dataIndex];
  // passing true you can get the final value of element property
  // at the end of animation
  const {width} = model.getProps(['width'], true); 
  console.log('width: ' + width); 
  return 4;
}
```
